suite/tests/linux/signal_pre_syscall.c

```diff
@@ -62,59 +58,125 @@ fail(const char *s)
     exit(1);
 }
 
+/* The signal handler copies one global to the other so that we can
+ * discover when the signal happened.
+ */
 void
-handler(int signum, siginfo_t *info, void *ucontext_)
+handler(int signum, siginfo_t *info, void *ucontext)
 {
-    /* We just need the handler to exist; it needn't do anything but sigreturn. */
+    flag_dst = flag_src;
 }
 
-void
-setup(void)
+/* Return -1 if the signal happened before the critical block, 0 if it
+ * happened during the critical block, or 1 if it happened after the
+ * critical block or the timer was cancelled. Under DynamoRIO the
+ * signal will never appear to have happened during the critical block
+ * because we delay the delivery of async signals to the end of the block.
+ */
+int
+try(timer_t timer, unsigned long long timer_duration_ns)
 {
-    struct sigaction act;
-    struct sigevent sevp;
+    flag_dst = 1;
+    flag_src = -1;
+
+    /* Set timer. */
+    struct itimerspec spec = {
+        { 0, 0 }, { timer_duration_ns / 1000000000, timer_duration_ns % 1000000000 }
+    };
+    if (timer_settime(timer, 0, &spec, 0))
```

Comment:
s/0/nullptr/ for the 4th arg (old_value).

Reply:
Using NULL as this is C.